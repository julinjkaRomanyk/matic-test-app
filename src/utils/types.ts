export type InfoStore = {
    firstName: string;
    lastName: string;
    phoneNumber: string;
    stateName: string;
};

export type Progress = {
    current: number;
    total: number;
};

export enum Fields {
    nameQuestion = "nameQuestion",
    phoneQuestion = "phoneQuestion",
    stateQuestion = "stateQuestion",
};


export type DefaultOption = { 
    label: string; 
    value: string 
};