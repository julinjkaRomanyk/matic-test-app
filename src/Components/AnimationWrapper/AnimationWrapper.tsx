import React, { useEffect } from "react";

type Props = {
  show: boolean;
  children: React.ReactNode;
}

export const Fade: React.FC<Props> = ({ show, children }) => {
  const [shouldRender, setRender] = React.useState(show);

  useEffect(() => {
    if (show) setRender(true);
  }, [show]);

  const onAnimationEnd = () => {
    if (!show) setRender(false);
  };

  return (
    shouldRender ? (
      <div
        style={{ animation: `${show ? "fadeIn" : "fadeOut"} 0s` }}
        onAnimationEnd={onAnimationEnd}
        data-testid="animation-wrapper"
      >
        {children}
      </div>
    ) : null
  );
};
