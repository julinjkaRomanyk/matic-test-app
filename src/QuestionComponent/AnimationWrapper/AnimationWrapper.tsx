import React, { useEffect, useState } from "react";

type Props = {
  show: boolean;
  children: React.ReactNode;
}

export const Fade: React.FC<Props> = ({ show, children }) => {
  const [shouldRender, setRender] = useState(show);

  useEffect(() => {
    if (show) setRender(true);
  }, [show]);

  const onAnimationEnd = () => {
    if (!show) setRender(false);
  };

  return (
    shouldRender ? (
      <div
        style={{ animation: `${show ? "fadeIn" : "fadeOut"} 0s` }}
        onAnimationEnd={onAnimationEnd}
      >
        {children}
      </div>
    ) : null
  );
};
