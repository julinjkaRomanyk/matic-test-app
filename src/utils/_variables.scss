$BasicBlack: black;
$BasicOrange: #FF5122;
$BasicWhite: white;
$SecondOrange: #FFBE1E;
$BackgroundColor: #FCFAF9;
$Grey: #726969;
$BorderColor: #AFA7A6;
$DarkPowder: #F9ECE7;
