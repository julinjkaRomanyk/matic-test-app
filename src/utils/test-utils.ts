export function getTestName(dirname: string = __dirname) {
    return dirname.split("/").reverse()[0];
  };