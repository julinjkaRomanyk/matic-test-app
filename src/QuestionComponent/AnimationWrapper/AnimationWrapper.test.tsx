import React from "react";
import { getTestName } from "../../utils/test-utils";
import { render, fireEvent } from "@testing-library/react";

import { Fade } from "./AnimationWrapper";

const inputToRender = () => {
    return (
        <>
             <span>Hello</span>
        </>
        )
    };

describe(getTestName(__dirname), () => {
    test("Match snapshot", () => {
      const { container } = render(<Fade show={true} children={inputToRender} />);
  
      expect(container.firstChild).toMatchSnapshot();
    });
  });
  
