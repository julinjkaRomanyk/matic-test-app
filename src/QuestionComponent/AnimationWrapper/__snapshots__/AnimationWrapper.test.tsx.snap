// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimationWrapper Match snapshot 1`] = `
<div
  style="animation: fadeIn 0s;"
/>
`;
